Aws 19.1.0 (#1147)
giantswarm/releases: aws/v19.1.0/README.md

# :zap: Giant Swarm Release v19.1.0 for AWS :zap:

This is a maintenance release featuring latest 1.24 Kubernetes versions as well as components upgrades. This release also introduces new features which are described in next sections.

***IAM Permissions Requirements***
The minimal requirement for the IAM permissions is [Version 3.3.0](https://github.com/giantswarm/giantswarm-aws-account-prerequisites/blob/master/CHANGELOG.md#330---2023-05-11) of [giantswarm-aws-account-prerequisites](https://github.com/giantswarm/giantswarm-aws-account-prerequisites/) repository.

## Cilium AWS ENI mode

Following our work on changing the CNI of the Giant Swarm Workload Clusters towards Cilium, we have added a possibility to migrate to the Cilium AWS ENI mode instead of plain Cilium setup.

> > **WARNING:** The Cilium AWS ENI mode can be *ONLY* enabled while upgrading from `18.4.2` to `19.1.0` release. From that point forward the Workload Clusters will be running in Cilium AWS ENI mode and cannot be switched back to our default Cilium that comes with `19.0.0`. Both the Cilium and Cilium AWS ENI mode will receive the same level of support going forward.

Reviewer: ```suggestion
> **WARNING:** The Cilium AWS ENI mode can *ONLY* be enabled while upgrading from `18.4.2` to `19.1.0` release. From that point forward the Workload Clusters will be running in Cilium AWS ENI mode and cannot be switched back to our default Cilium that comes with `19.0.0`. Both the Cilium and Cilium AWS ENI mode will receive the same level of support going forward.
```

> This feature can be enabled via an annotation `cilium.giantswarm.io/ipam-mode: eni` set on the Cluster CR while on `18.4.2` release prior to `19.1.0` upgrade. When the upgrade is triggered, the underlying infrastructure will choose to continue with the [Cilium AWS ENI mode](https://docs.cilium.io/en/latest/network/concepts/ipam/eni/). This is meant for the users that do not want to migrate any of the underlying network infrastructure that has been linked with the Giant Swarm Workload Clusters. The network setup after the upgrade will be the same as while running `aws-cni` with `kube-proxy`.

Reviewer: ```suggestion
This feature can be enabled via the annotation `cilium.giantswarm.io/ipam-mode: eni`, set on the Cluster CR, while on `18.4.2` release and prior to `19.1.0` upgrade. When the upgrade is triggered, the underlying infrastructure will choose to continue with the [Cilium AWS ENI mode](https://docs.cilium.io/en/latest/network/concepts/ipam/eni/). This is meant for the users that do not want to migrate any of the underlying network infrastructure that has been linked with the Giant Swarm Workload Clusters. The network setup after the upgrade will be the same as while running `aws-cni` with `kube-proxy`.
```

### kubernetes [1.24.17](https://github.com/kubernetes/kubernetes/releases/tag/v1.24.17)

#### Feature
- Kubernetes is now built with Go 1.20.7 ([#119837](https://github.com/kubernetes/kubernetes/pull/119837), [@jeremyrickard](https://github.com/jeremyrickard)) [SIG Apps, Cloud Provider, Node, Release, Storage and Testing]
#### Bug or Regression
- Fixed a bug where clusters that use KMS v1 with skewed API servers on versions v1.24 and v1.25 would see internal errors when attempting to read encrypted data via the v1.24 API servers. ([#119387](https://github.com/kubernetes/kubernetes/pull/119387), [@enj](https://github.com/enj)) [SIG API Machinery and Auth]
#### Dependencies
#### Added
_Nothing has changed._
#### Changed
_Nothing has changed._
#### Removed
_Nothing has changed._

### app-operator [6.8.0](https://github.com/giantswarm/app-operator/releases/tag/v6.8.0)

#### Added
- Add Service Monitor by default to make it complain with the latest monitoring improvements 

### aws-operator [14.21.0](https://github.com/giantswarm/aws-operator/releases/tag/v14.21.0)

### cert-manager [3.3.0](https://github.com/giantswarm/cert-manager-app/releases/tag/v3.3.0)

> ⚠️ Attention: Major release [3.0.0](#300---2023-07-26) contains breaking changes in user values! Please make yourself familiar with its changelog! ⚠️
> #### Added

Reviewer 2: ```suggestion
⚠️ Attention: Major release [3.0.0](#300---2023-07-26) contains breaking changes in user values! Please make sure you [read about the upgrade instructions](https://github.com/giantswarm/cert-manager-app/blob/main/docs/upgrading.md)! ⚠️
#### Added
```

## Kyverno by default

This is the release preparing for the migration away from Pod Security Policies (PSP) in favor of Pod Security Standards (PSS) in Kubernetes 1.25. Our `security-bundle` is now installed by default, and will deploy `kyverno` and `restricted` level PSS policies in `audit` mode. These resources are provided in order to allow time to create exceptions for workloads which need them before the policies are changed to `enforce` in a future release. For more information about PSS please read our official [documentation](https://docs.giantswarm.io/advanced/security-policy-enforcement/). Please also take a look at the `kyverno` [documentation](https://docs.giantswarm.io/platform-overview/security/platform-security/#kyverno) to utilize fully its potential.

> > **WARNING:** If you are already running `kyverno` as Giant Swarm Managed App, the installation of `security-bundle` will fail. However the already existing `kyverno` deployment and its configuration can be adopted by the bundle after the upgrade is finished. Please talk to your Account Engineer if you have any questions.

Reviewer 3: I would probably ask the Account Engineers to be proactive about it considering how it went with the prometheus operator @T-Kukawka  :)

Reviewer 4: This wasn't that bad as it can be `adopted` before but yes that should be in the checklist for usre with @giantswarm/team-shield

Reviewer 5: If it's simple, instructions can be included here?!

Reviewer 6: We're already in contact with customers using the bundle and have migration plans with them

This feature allows customer to set an annotation on AWSMachineDeployment CR's to change the Flatcar Release Version. For now it only allows to set `alpha.giantswarm.io/flatcar-release-version: "3689.0.0"` or a higher. We have added this feautre to accomodate the issues with Cilium CNI high CPU usage on small clusters. This feature is solely to enable customers to run Flatcar `alpha` channel which consists of the `kernel 6` version that fixes the issue, while waiting for a `stable` Flatcar release.

Please read the detailed description of the designed behaviour of the annotation:
- when setting the annotation the TCNP Stack for the specific node pool is rolled and replaces the OS image
- when removing the annotation the node pool is updated and switches back to the default OS image which is coming from the AWS release
- when upgrading the cluster to a new AWS release, the node pool uses the specific flatcar release from the annotation as long as you don't change by either setting it to a higher version or removing the annotation.

## Change details

### app-operator [6.8.0](https://github.com/giantswarm/app-operator/releases/tag/v6.8.0)

#### Added

> - Add Service Monitor by default to make it complain with the latest monitoring improvements 

Reviewer 3: ```suggestion

- Add Service Monitor by default to make it compliant with the latest monitoring improvements 
```

### aws-operator [14.21.0](https://github.com/giantswarm/aws-operator/releases/tag/v14.21.0)

#### Added
- Allow newer flatcar releases for node pools as provided by AWS release.
- Add sigs.k8s.io/cluster-api-provider-aws/role tag to all subnets as preparation for migration to CAPI.

#### Changed
- Unmanage interfaces for CNI eth[1-9] on workers eth[2-9] on masters
- [cilium eni mode] Only run aws-node, calico and kube-proxy on old nodes during migration to cilium.

### cert-operator [3.2.1](https://github.com/giantswarm/cert-operator/releases/tag/v3.2.1)

#### Fixed
- Fix rule names of PolicyException.

### cluster-operator [5.8.0](https://github.com/giantswarm/cluster-operator/releases/tag/v5.8.0)

#### Added
- Add ENI mode for Cilium on AWS.
- Consider new control-plane label.
#### Changed
- Propagate `global.podSecurityStandards.enforced` value set to `true` for PSS migration

> - Rename function for better readbility.

Reviewer 3: ```suggestion
- Rename function for better readability.
```

# :zap: Giant Swarm Release v19.1.0 for AWS :zap:

This is a maintenance release featuring latest 1.24 Kubernetes versions as well as components upgrades. This release also introduces new features which are described in next sections.

***IAM Permissions Requirements***
The minimal requirement for the IAM permissions is [Version 3.3.0](https://github.com/giantswarm/giantswarm-aws-account-prerequisites/blob/master/CHANGELOG.md#330---2023-05-11) of [giantswarm-aws-account-prerequisites](https://github.com/giantswarm/giantswarm-aws-account-prerequisites/) repository.

## Cilium AWS ENI mode

Following our work on changing the CNI of the Giant Swarm Workload Clusters towards Cilium, we have added a possibility to migrate to the Cilium AWS ENI mode instead of plain Cilium setup.

> **WARNING:** The Cilium AWS ENI mode can be *ONLY* enabled while upgrading from `18.4.2` to `19.1.0` release. From that point forward the Workload Clusters will be running in Cilium AWS ENI mode and cannot be switched back to our default Cilium that comes with `19.0.0`. Both the Cilium and Cilium AWS ENI mode will receive the same level of support going forward.

> Feature can be enabled via an annotation `cilium.giantswarm.io/ipam-mode: eni` set on the Cluster CR while on `18.4.2` release prior to `19.1.0` upgrade. When the upgrade is triggered, the undelying infrastructure will choose to continue with the [Cilium AWS ENI mode](https://docs.cilium.io/en/latest/network/concepts/ipam/eni/). This is meant for the users that do not want to migrate any of the underlying network infrastructure that has been linked with the Giant Swarm Workload Clusters. The network setup after the upgrade will be the same as while running `aws-cni` with kube-proxy.

Reviewer 4: fixed

## Kyverno by default

This is the release preparing for the migration away from Pod Security Policies (PSP) in favor of Pod Security Standards (PSS) in Kubernetes 1.25. Our `security-bundle` is now installed by default, and will deploy `kyverno` and `restricted` level PSS policies in `audit` mode. These resources are provided in order to allow time to create exceptions for workloads which need them before the policies are changed to `enforce` in a future release. For more information about PSS please read our official [documentation](https://docs.giantswarm.io/advanced/security-policy-enforcement/). Please also take a look at the `kyverno` [documentation](https://docs.giantswarm.io/platform-overview/security/platform-security/#kyverno) to utilize fully its potential.

> **WARNING:** If you are already running `kyverno` as Giant Swarm Managed App, the installation of `security-bundle` will fail. However the already existing `kyverno` deployment and its configuration can be adopted by the bundle after the upgrade is finished. Please talk to your Account Engineer if you have any questions.

## AWSMachineDeployment CR's annotation to change the Flatcar Release Version

> This feature allows customer to set an annotation on AWSMachineDeployment CR's to change the Flatcar Release Version. For now it only allows to set `alpha.giantswarm.io/flatcar-release-version: "3689.0.0"` or a higher. We have added this feautre to accomodate the issues with Cilium CNI high CPU usage on small clusters. This feature is solely to enable customers to run Flatcar `alpha` channel which consists of the `kernel 6` version that fixes the issue, while waiting for a `stable` Flatcar release.

Reviewer 5: ```suggestion
This feature allows customers to set an annotation on AWSMachineDeployment CR's to change the Flatcar Release Version. For now it only allows setting `alpha.giantswarm.io/flatcar-release-version: "3689.0.0"` or a higher version. We have added this feature to accommodate the issues with Cilium CNI high CPU usage on small clusters. This feature is solely to enable customers to run the Flatcar `alpha` channel which consists of the `kernel 6` version that fixes the issue, while waiting for a `stable` Flatcar release.
```

> This feature allows customer to set an annotation on AWSMachineDeployment CR's to change the Flatcar Release Version. For now it only allows to set `alpha.giantswarm.io/flatcar-release-version: "3689.0.0"` or a higher. We have added this feautre to accomodate the issues with Cilium CNI high CPU usage on small clusters. This feature is solely to enable customers to run Flatcar `alpha` channel which consists of the `kernel 6` version that fixes the issue, while waiting for a `stable` Flatcar release.

Reviewer 4: fixed

> Please read the detailed description of the designed behaviour of the annotation:

Reviewer 5: ```suggestion
The annotation behaves as follows:
```

Otherwise "Please read" sounds like there should be a hyperlink to documentation.

> - when setting the annotation the TCNP Stack for the specific node pool is rolled and replaces the OS image

Reviewer 5: ```suggestion
- when setting the annotation, the TCNP CloudFormation Stack for the specific node pool is rolled and replaces the OS image
```

> - when removing the annotation the node pool is updated and switches back to the default OS image which is coming from the AWS release

Reviewer 5: ```suggestion
- when removing the annotation, the node pool is updated and switches back to the default OS image which is coming from the AWS release
```

> - when upgrading the cluster to a new AWS release, the node pool uses the specific flatcar release from the annotation as long as you don't change by either setting it to a higher version or removing the annotation.

Reviewer 7: What if a customer fixed the Flatcar version to (imagine) 1.23.4 while the new AWS release requires 1.34.5, is not compatible with the former and would break the cluster during upgrade?

Reviewer 8: You can only take the default flatcar version from the release or upgrade to 3689.0.0 or higher. However we have a JSON file for choosing the right AMI for every AWS region. If the flatcar version isn't found you can't to upgrades, meaning you can't accidentally select a wrong flatcar release

Reviewer 7: You might have gotten me wrong: Imagine being on a AWS release and having defined your own Flatcar version for a node pool. You then want to upgrade to another AWS release and still have defined your own Flatcar version in the node pool, but the new AWS release is not compatible with this Flatcar version.

Are we then falling back to the default Flatcar version shipped with the new AWS release or is the whole upgrade being stopped/prevented?

Reviewer 4: It will have to be reviewed before every upgrade, what Flatcar version there is etc, but in general it happens very rarely that they are incompatible. Also u change for 'higher' release to which we most probably will upgrade hence the risk is small.

Reviewer 7: Thanks a lot for the feedback! :) Sounds good to me, so we only need to take care of making customers aware of it.

Reviewer 4: yes, this feature is targeted for 1 customer only for now

- Adjusted the admission controller to give it more QPS against the API
- Adjusted the updater to give it more QPS against the API
- Adjusted the recommender to give it
  - more QPS against the API
  - doubling the memory in case of an OOMKilled event
  - Using the 95% percentile for the calculation of the CPU usage: should allow to scale up more precisely to account for spikes in CPU consumption of the workload
  - Adjusted the resource and limits to accomodate larger clusters by default
  - Calculating recommendations only for workloads which do have a VPA custom resource, instead of all workloads
  - Removed standard network policies to decrease maintenance burden
  - Fixed Cilium Network Policy to allow CRD jobs execution
  - Added Cilium Network Policy weight for an early execution
  - Disabled VPA for the updater pod otherwise it keeps on getting re-scheduled because the memory consumption varies a lot between reconsiling resources and idle
  - Disabled VPA for the recommender pod otherwise it keeps on getting re-scheduled because the memory consumption varies a lot between reconsiling resources and idle

> ### observability-bundle [0.8.2](https://github.com/giantswarm/observability-bundle/releases/tag/v0.8.)

Reviewer 9: ```suggestion
### observability-bundle [0.8.2](https://github.com/giantswarm/observability-bundle/releases/tag/v0.8.2)
```

Reviewer 4: thanks! <3

### kubernetes [1.24.17](https://github.com/kubernetes/kubernetes/releases/tag/v1.24.17)

#### Feature
- Kubernetes is now built with Go 1.20.7 ([#119837](https://github.com/kubernetes/kubernetes/pull/119837), [@jeremyrickard](https://github.com/jeremyrickard)) [SIG Apps, Cloud Provider, Node, Release, Storage and Testing]
#### Bug or Regression
- Fixed a bug where clusters that use KMS v1 with skewed API servers on versions v1.24 and v1.25 would see internal errors when attempting to read encrypted data via the v1.24 API servers. ([#119387](https://github.com/kubernetes/kubernetes/pull/119387), [@enj](https://github.com/enj)) [SIG API Machinery and Auth]
#### Dependencies
#### Added
_Nothing has changed._
#### Changed
_Nothing has changed._
#### Removed
_Nothing has changed._

### app-operator [6.8.0](https://github.com/giantswarm/app-operator/releases/tag/v6.8.0)

#### Added
- Add Service Monitor by default to make it complain with the latest monitoring improvements 

### aws-operator [14.22.0](https://github.com/giantswarm/aws-operator/releases/tag/v14.22.0)

#### Changed

- Get AMI data from helm value rather than from hardcoded string in the code.

> ⚠️ Attention: Major release [3.0.0](#300---2023-07-26) contains breaking changes in user values! Please make sure you [read about the upgrade instructions](https://github.com/giantswarm/cert-manager-app/blob/main/docs/upgrading.md)! ⚠️

Reviewer 10: why do we have `cert-manager` stuff under `aws-operator`?

Reviewer 4: fixed, maybe smbd messed up copy paste
